app/lib/katello/resources/candlepin/upstream_consumer.rb

```diff
@@ -22,6 +22,8 @@ def get(params)
             JSON.parse(super(path + hash_to_query(params) + includes, self.default_headers).body)
           rescue RestClient::Gone
             raise ::Katello::Errors::UpstreamConsumerGone
+          rescue RestClient::NotFound
```

Comment:
This should actually be handled in the `ping` method of this class. See how `UpstreamConnectionChecker` uses it

Also, is there any test coverage to be updated?